test/sub.js

```diff
@@ -14,3 +15,11 @@ test('create many subs', async function (t) {
 
   t.is(count, root.core.listenerCount('append'))
 })
+
+test('sub encoding', async function (t) {
+  const db = create()
+  const keyEncoding = new SubEncoder('files', 'utf-8')
+
+  await db.put('/a', '1', { keyEncoding })
+  t.alike(await db.get('/a', { keyEncoding }), { seq: 1, key: '/a', value: '1' })
```

Comment:
Is it correct that it's giving a buffer as key? Instead of the string '/a'

Reply:
Upgraded sub-encoder v1 to v2, it seems to pass correctly the test above!

Now this test is not passing with the new `encodeRange` https://github.com/holepunchto/hyperbee/blob/55b9e00df7a4f3c4d7f585bf62d0d57a2fbfead0/test/watch.js#L665

Anyway at least sub-encoder was working as intended!